server.cc

```diff
@@ -1,265 +1,91 @@
-  auto const port = static_cast<uint16_t>(std::atoi("8888"));
-  auto const threads = std::max<int>(1, std::atoi("16"));
-
-  // The io_context is required for all I/O
-  boost::asio::io_context ioc{threads};
-
-  // Create and launch a listening port
-  std::make_shared<listener>(ioc, tcp::endpoint{address, port})->run();
-  std::cout << "Server started..."
-            << "\n";
-
-  // Run the I/O service on the requested number of threads
-  std::vector<std::thread> v;
-  v.reserve(threads - 1);
-  for (auto i = threads - 1; i > 0; --i) v.emplace_back([&ioc] { ioc.run(); });
-  ioc.run();
-
-  return EXIT_SUCCESS;
+#include <served/plugins.hpp>
+#include <served/served.hpp>
+#include "./apis.h"
+#include "./config.h"
+#include "cJSON.h"
+
+int main(int, char const**) {
+  served::multiplexer mux;
+  mux.use_after(served::plugin::access_log);
+
+  iota_client_service_t service;
+  service.http.path = "/";
+  service.http.host = IRI_HOST;
+  service.http.port = IRI_PORT;
+  service.http.api_version = 1;
+  service.serializer_type = SR_JSON;
+  iota_client_core_init(&service);
+
+  /**
+   * @method {get} /tag/:tag Find transactions by tag
+   *
+   * @param {String} tag Must be 27 trytes long
+   *
+   * @return {String[]} hashes List of transaction hashes
+   */
+  mux.handle("/tag/{tag:[A-Z9]{27}}")
+      .get([&](served::response& res, const served::request& req) {
+        cJSON* json_obj = cJSON_CreateObject();
+        cJSON_AddStringToObject(json_obj, "tag", req.params["tag"].c_str());
+        const char* json = cJSON_PrintUnformatted(json_obj);
+        char* json_result;
+
+        api_find_transactions_by_tag(&service, json, &json_result);
+        res.set_header("content-type", "application/json");
+        res << json_result;
+
+        cJSON_Delete(json_obj);
+      });
+
+  /**
+   * @method {get} /tips Fetch all tips
+   *
+   * @return {String[]} tips List of transaction hashes
+   */
+  mux.handle("/tips").get(
+      [&](served::response& res, const served::request& req) {
+        char* json_result;
+
+        api_get_tips(&service, &json_result);
+        res.set_header("content-type", "application/json");
+        res << json_result;
+      });
+
+  /**
+   * @method {get} /address Generate an unused address
+   *
+   * @return {String} hash of address hashes
+   */
+  mux.handle("/address")
+      .get([&](served::response& res, const served::request& req) {
+        char* json_result;
+
+        api_generate_address(&service, &json_result);
+        res.set_header("content-type", "application/json");
+        res << json_result;
+      });
+
+  /**
+   * @method {get} / Client bad request
+   *
+   * @return {String} message Error message
+   */
+  mux.handle("/").get([](served::response& res, const served::request&) {
+    cJSON* json_obj = cJSON_CreateObject();
+    cJSON_AddStringToObject(json_obj, "message", "Invalid path");
+    const char* json = cJSON_PrintUnformatted(json_obj);
+
+    res.set_status(400);
```

Comment:
Can you enumerate the possible error codes?

Reply:
According to [Tangle-Accelerator Core API](https://hackmd.io/s/Skn9TizTm). We have `400` for client bad request and `404` for request not found.

Reply:
Let's introduce macro(s) to map these error codes.